dags/aggregates_python_helpers.py

```diff
@@ -143,13 +149,20 @@ def get_first_day_of_previous_month(date_str):
     new_date = datetime(year, month, 1)
     return new_date.strftime('%Y-%m-%d %H:%M:%S')  # Return the new date as a string in the same format
 
-def insert_permissions_to_db(df, db_params):
+def insert_permissions_to_db(df):
     """Load data from a Pandas DataFrame into the database in batches after filtering out null dates."""
 
     batch_size=1000
 
-    # Create a database engine
-    engine = create_engine(f"postgresql://{db_params['user']}:{db_params['password']}@{db_params['host']}/{db_params['dbname']}")
+    # Establish a BigQuery connection
+    client = bigquery.Client()
+    table_id = "airflow-lab-415614.airflow_dataset.reporting_results2020"
```

Comment:
in a lot of places you are using the dataset,schema and table name. maybe we could consider using the https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/params.html

Reply:
ok, I've got a question to this - Will it work efficiently? I've got a feeling (correct me if I'm wrong) that applying this change will still require setting all the connection parameters in the related functions' bodies (this time, with a use of calling these DAG params), and the only difference and feature of this will be the fact that the params will be able to be changed in one place, by DAG params declaration.
Is this what you mean?